### DISCUSSION TITLE
WebView2 production support for UWP apps

### DISCUSSION THREAD

**Discussion by radujipa** [microsoft/WindowsAppSDK#752] [Q&A] [answered]
Hello everyone! 👋 

I couldn't find any references to when WebView2 (through WinUI3) will be available to production UWP apps published on the Store. It would be useful to know the latest estimates around this as we've been following the developements of this since last year and were expecting to upgrade our apps at beginning of Q2 this year.

Many thanks for your time! 🙏

**Comment by andrewleader** [comment, marked as answer]
Hi @radujipa, thanks for asking. We currently don't have a timeline/date. Preview (non-production) support for UWP apps will be back in the upcoming 0.8 preview VSIX's. We're actively working on locking down our plan and will share details when we have them.

**Reply by radujipa** [reply]
Thanks for the answer @andrewleader, I can appreciate there being a large number of moving parts with this. Our apps primarily rely on WebView to serve locally builtin web apps and then make use of UWP APIs to add native integrations (notifications, live tiles, OEM support, etc). We're only distributing these through the Microsoft Store and require compatibility with S-mode devices.

We've therefore been excited about upgrading to WebView2 across the board however it's a little difficult at the moment to understand the recommended path to doing so. Would we have to migrate existing code to Project Reunion in order to do so?

**Reply by andrewleader** [reply]
If you're looking to migrate toa desktop-based WebView2 app now, I don't think we have formal docs on that, but here's a high-level...

You would...

1. Create a new WinUI 3 Desktop app (using Project Reunion VSIX)
2. Copy your existing code (XAML namespaces change from Windows.* to Microsoft.*), most UWP APIs will work too but some like Secondary Tiles might require some changes since they're being used from a desktop MSIX app rather than UWP.

So, it's currently definitely a migration... not a full rewrite but can be painful especially if you're depending on other libraries that haven't switched to WinUI 3. But the hope is that 95% of your code should port over with just simple namespace changes, and then a few APIs will need some modifications, and a few UWP APIs are just simply not supported by desktop MSIX apps.

We'll be documenting this more over time!
